jetstream/pull.go

```diff
@@ -185,16 +185,21 @@ func (p *pullConsumer) Consume(handler MessageHandler, opts ...PullConsumeOpt) (
 			return
 		}
 		defer func() {
-			if sub.pending.msgCount < consumeOpts.ThresholdMessages ||
-				(sub.pending.byteCount < consumeOpts.ThresholdBytes && sub.consumeOpts.MaxBytes != 0) &&
+			sub.Lock()
+			pending := sub.pending
```

Comment:
Copying the value of pending because next we are using `resetPendingMsgs` and it locks in the function itself so I can't create a lock defer unlock in addition we are already in a defer so it might seems overly complicated to defer again.